### PR TITLE
Draft: Remove unused code related to csv

### PR DESCRIPTION
**Pull request by royfalk**
The file VegaPersonalities.csv isn't used.

@ministerofinformation / @Loki1950 - please let me know if you think this file should have existed and used recently. 

Code Changes:
- [x] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation

Issues:
- Game crashed twice. Opened https://github.com/vegastrike/Vega-Strike-Engine-Source/issues/716 for second crash.

Purpose:
- What is this pull request trying to do? Remove CSV code by either converting to JSON or deleting if not used.

### CODE REVIEW

**Review by Loki1950 — COMMENTED** [review]
not sure about the personality file but it may be related to the story line

File: engine/src/cmd/ai/aggressive.cpp [hunk 1]
```diff
@@ -173,7 +148,7 @@ static AIEvents::ElemAttrMap *getLogicOrInterrupt(string name,
         vsUMap<string, AIEvents::ElemAttrMap *> &mymap,
         int personalityseed) {
     string append = "agg";
-    static vsUMap<string, string> myappend = getAITypes();
+    static vsUMap<string, string> myappend;
```
Comment (BenjamenMeyer) [inline engine/src/cmd/ai/aggressive.cpp:151]: how will this get populated now?

**Review by stephengtuggy — CHANGES_REQUESTED** [review]
I would rather not make this change at this time. This seems like one of those "fences" that we don't fully understand the purpose of yet. Once we do, we can come back and revisit.

File: engine/src/cmd/ai/aggressive.cpp (hunk 1 above)
Comment (royfalk) [reply engine/src/cmd/ai/aggressive.cpp:151]: If you look at line 113-115, this is what's happening anyway. The file 'VegaPersonalities' doesn't exist in either VS or WCPU. I've trimmed the code but nothing's changed. At least, that was the intention.
I specifically asked @Loki1950, as he's our resident historian for such things. 
As I can't figure out from the code what they wanted to do and this appears to be a stub for some future feature, I vote to delete it. If/When we get to developing, we'll hopefully document it as well :)

**Comment by royfalk** [comment]
> 

I absolutely don't understand the purpose of this. However, it also isn't active yet and never will be. The only two ways to figure this out are to ask one of the original devs and hope he remembers after a decade or meditate on the entire AI code and hope it becomes clearer then.
As noted above, this code doesn't do anything if the file is missing.

File: engine/src/cmd/ai/aggressive.cpp (hunk 1 above)
Comment (stephengtuggy) [reply engine/src/cmd/ai/aggressive.cpp:151]: I just checked. The file 'VegaPersonalities.csv' does, in fact, exist in PWCU. Three copies of it. One under the 'ai' folder, another under the 'ai.easy' folder, and the third under the 'ai.hard' folder.

Comment (royfalk) [reply engine/src/cmd/ai/aggressive.cpp:151]: Huh. I actually checked in what I assumed was the repository. Is it something you're running on your computer or is it in the repo as well? If it is, can you point me to it?

Comment (stephengtuggy) [reply engine/src/cmd/ai/aggressive.cpp:151]: https://github.com/pwcu/privateer_wcu/blob/master/ai.easy/VegaPersonalities.csv

**Comment by Loki1950** [comment]
I have a possible reason for it's existence as @stephengtuggy has mentioned it holds the AI profiles of NPC's in the story line since Upon a the Coldest Sea is just a stub ATM that file is not needed yet as it is in Privateer which does have a full story.

**Comment by royfalk** [comment]
Based on the example in the link above, I'll start working on converting it instead of simply deleting the whole function.

**Comment by royfalk** [comment]
OK. So I looked at the PU data and tried to make it work for VS and it crashed nicely. Looks like I'm going to have to run PU and debug that.

**Comment by royfalk** [comment]
I'm going to close this. The code is needed and I may as well start from scratch. There's nothing here to salvage.

**Pull request closed without merging**